api/v1alpha1/clienttrafficpolicy_types.go

```diff
@@ -200,6 +204,23 @@ type HTTP10Settings struct {
 	UseDefaultHost *bool `json:"useDefaultHost,omitempty"`
 }
 
+// HTTP2Settings provides HTTP/2 configuration on the listener.
+type HTTP2Settings struct {
+	// InitialStreamWindowSize sets the initial window size for HTTP/2 streams.
```

Comment:
can we create a custom type for `resource.Quantiy` defined in https://github.com/envoyproxy/gateway/blob/2c4d385fcfb02841483cc36e0292175f60249bbb/api/v1alpha1/connection_types.go#L24 and reuse it here ?

Reply:
kubebuilder won't generate validation for alias type. If we define a new type, we need to redefine every `resource.Quantiy` function that EG used for the new type `Size`.


https://github.com/envoyproxy/gateway/actions/runs/8772985239/job/24072530543?pr=3222